docs/proposals/2841-flux-hpc/README.md

```diff
@@ -0,0 +1,275 @@
+    *   A **ConfigMap** is generated containing two scripts: `init.sh` (for the init container) and `entrypoint.sh` (for the main container). This ConfigMap is mounted into the pods.
+4.  **Execution Wrap**: The command of the user's main application container is overridden to use the `entrypoint.sh` script. This script first sets up the Flux environment (using the files from the shared volumes) and then uses `flux start` and `flux submit` to launch the user's original command within the now-active Flux cluster.
+5.  **Networking**: The plugin ensures the `JobSet` is configured with a headless service and DNS hostnames enabled, which Flux uses for its broker communication. High speed network for MPI can be used by way of extending pods to use a bypass mechanism to support Infiniband (Azure) or the Elastic Fabric Adapter (AWS).
+
+This approach provides an HPC environment without requiring the user to build a custom Docker image with Flux pre-installed, significantly improving the user experience.
+
+### API Design
+
+The proposed changes will integrate into the existing `v1alpha1` API structures. We will add a new field, `hpc`, to the `MLPolicySource` struct. This aligns Flux with other runtimes like Torch and MPI.
+
+#### 1. `ClusterTrainingRuntime` and `TrainingRuntime`
+
+The proposal will leverage the existing `ClusterTrainingRuntime` (cluster-scoped) and `TrainingRuntime` (namespace-scoped) CRDs. These objects serve as reusable templates, and no changes are needed for their top-level structure.
+
+#### 2. Enhancing `MLPolicySource`
+
+We will add the `HPC` field to the `MLPolicySource` struct, which is embedded within the `MLPolicy`.
+
+```go
+// MLPolicySource is the runtime-specific configuration
+// One of the following specs can be set.
+type MLPolicySource struct {
+    // torch defines the configuration for the PyTorch runtime.
+    // +optional
+	Torch *TorchMLPolicySource `json:"torch,omitempty"`
+
+	// mpi defines the configuration for the MPI Runtime.
+	// +optional
+	MPI *MPIMLPolicySource `json:"mpi,omitempty"`
+
+	// hpc defines the configuration for an hpc runtime (e.g., Flux Framework).
+	// This is the new field being added.
+	// +optional
+	HPC *HPCMLPolicySource `json:"hpc,omitempty"`
+}
+
+// HPCMLPolicySource represents an HPC runtime configuration.
+type HPCMLPolicySource struct {
+	// Manager specifies the workload manager to use.
+	// +kubebuilder:default="flux"
+	// +optional
+	Manager string `json:"manager,omitempty"`
+
+	// Tasks is the number of tasks to provide to the workload manager.
+	// +optional
+	Tasks *int32 `json:"tasks,omitempty"`
```

Comment:
Why do we need to pre-define number of Tasks?

Reply:
Yes that's exactly it.

Reply:
I would prefer to use `numProcPerNode` field if we go towards `FluxPolicy` API, so we will be consistent with MPI and Torch.

Reply:
Resolving here - we are doing `numProcPerNode` to be consistent with others.